tests/test_integration.py

```diff
@@ -27,5 +31,77 @@ def test_smoke(self):
         demo_file = os.path.join(project_dir, 'demo.py')
         subprocess.check_output([sys.executable, demo_file])
 
+    def test_scenario_setup(self):
+
+        w, h = random.randint(10,20), random.randint(10,20)
+        num_poi, num_cs, num_new_cs = (random.randint(1,4), random.randint(1,4), random.randint(1,4))
+
+        G, pois, charging_stations, potential_new_cs_nodes = demo.set_up_scenario(w, h, num_poi, num_cs)
+
+        self.assertEqual(len(G.nodes()), w*h)
+        self.assertEqual(len(pois), num_poi)
+        self.assertEqual(len(charging_stations), num_cs)
+        self.assertEqual(len(potential_new_cs_nodes), len(G.nodes())-len(charging_stations))
+
+    def test_num_new_cs(self):
+        """Check that correct number of new charging locations are found in a random scenario"""
+
+        w, h = (random.randint(10,20), random.randint(10,20))
+        num_poi, num_cs, num_new_cs = (random.randint(1,4), random.randint(1,4), random.randint(1,4))
+
+        G, pois, charging_stations, potential_new_cs_nodes = demo.set_up_scenario(w, h, num_poi, num_cs)
+
+        bqm = demo.build_bqm(potential_new_cs_nodes, num_poi, pois, num_cs, charging_stations, num_new_cs)
+
+        sampler = neal.SimulatedAnnealingSampler()
+        new_charging_nodes = demo.run_bqm_and_collect_solns(bqm, sampler, potential_new_cs_nodes)
```

Comment:
You're going to need to update these to run_bqm_and_collect_solutions() as well

Reply:
Good catch, thanks!